src/main/java/com/codedx/burp/ExportActionListener.java

```diff
@@ -0,0 +1,132 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.codedx.burp;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.mime.HttpMultipartMode;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.FileBody;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import burp.BurpExtender;
+import burp.IBurpExtenderCallbacks;
+import burp.IScanIssue;
+
+public class ExportActionListener implements ActionListener{
+	private BurpExtender burpExtender;
+	private IBurpExtenderCallbacks callbacks;
+	
+	public ExportActionListener(BurpExtender be, IBurpExtenderCallbacks cb){
+		burpExtender = be;
+		callbacks = cb;
+	}
+	
+	public void actionPerformed(ActionEvent e){
+		IScanIssue[] issues = getIssues();
+		if(issues != null && issues.length > 0){
+			File report = new File("burp_codedx-plugin.xml");
+			callbacks.generateScanReport("XML", issues, report);
+			if(report != null && report.exists()){
+				try{
+					HttpResponse response = sendData(report, getServer());
+					StatusLine responseLine = response.getStatusLine();
+					int responseCode = responseLine.getStatusCode();
+					if(responseCode == 202){
+						burpExtender.message("The report was successfully uploaded to Code Dx.", "Success");
+					} else if(responseCode == 400) {
+						burpExtender.error("An unexpected error occurred and the report could not be sent.\nThe server returned Error 400: Bad Request" + getResponseError(response));
+					} else if(responseCode == 403){
+						burpExtender.error("The report could not be sent. The server returned Error 403: Forbidden.\nThe API Key may be incorrect or have insufficient permissions for this project.");
+					} else if(responseCode == 404){
+						burpExtender.error("The report could not be sent. The server returned Error 404: Not Found.\nThe Server URL may be wrong or the project may no longer exist.");
+					} else if(responseCode == 415) {
+						burpExtender.error("An unexpected error occurred and the report could not be sent.\nThe server returned Error 400: Bad Request" + getResponseError(response));
```

Comment:
The message for code 415 says the code was 400. 415 is for when the file couldn't be identified as anything useful for analyisis (this could possibly happen if the burp report format changes to something that Code Dx can't recognize, or if the report is broken for some reason).